src/mongo_logger.ts

```diff
@@ -415,10 +416,15 @@ export function stringifyWithMaxLen(
 ): string {
   let strToTruncate = '';
 
-  if (typeof value === 'function') {
-    strToTruncate = value.toString();
-  } else {
-    strToTruncate = EJSON.stringify(value, options);
+  try {
+    strToTruncate =
+      typeof value !== 'function'
+        ? EJSON.stringify(value, options)
+        : value.name !== ''
+        ? value.name
+        : 'anonymous function';
```

Comment:
Could you determine when we're passing in a function here? This is for server selection right? "anonymous function" is not going to be very descriptive so we should make a note of what functions are coming in here and give them informative names.

Reply:
So can you identify when you are seeing an anonymous function enter this code? like using the debugger or print stmt when you see an empty function name. Then we can trace back to the input function and find out why it doesn't have a name and give it one, possibly in a followup.

Reply:
A user can pass in an anonymous/ nameless function as their serverSelector:

```
myTopology.selectServer(
    function (topologyDescription: TopologyDescription, servers: ServerDescription[], deprioritized?: ServerDescription[]): ServerDescription[] { .. do something }, 
    {}, 
    callback
);
```

I've included a test case for this as well under `describe('stringifyWithMaxLen', function ...`

Reply:
Bookkeeping: We've clarified that the logger is not public / that the user cannot access it. 

We call server selection with anonymous functions [in execute operation](https://github.com/mongodb/node-mongodb-native/blob/7f97c2a2aeb88d7ee0394ade258e440ddfcb3b2f/src/operations/execute_operation.ts#L142). (When [secondaryWritableServerSelector]( https://github.com/mongodb/node-mongodb-native/blob/7f97c2a2aeb88d7ee0394ade258e440ddfcb3b2f/src/sdam/server_selection.ts#L57) or [sameServerSelector](https://github.com/mongodb/node-mongodb-native/blob/7f97c2a2aeb88d7ee0394ade258e440ddfcb3b2f/src/sdam/server_selection.ts#L39) are called)

Reply:
Can we file a ticket to give those functions names (or do that here?)